### DISCUSSION TITLE
Can the grid/clips in the TritonRoute be exported?

### DISCUSSION THREAD

**Discussion by Mohamed-Mejri** [The-OpenROAD-Project/OpenROAD#1732] [Q&A] [answered]
Hello, 
I would like to know if the grid can be exported after the processing of the global router output and/or the TA step.

I'm planning on trying some other AI based approaches instead of Heuristics. And if I can export the grid or the clips it would be more helpful.

Any idea/help will be much appreciated!

**Comment by maliberty** [comment]
Do you mean the gcell grid?  There are messages

"GCELLGRID X {} DO {} STEP {} ;",
"GCELLGRID Y {} DO {} STEP {} ;",

that will give you those values.  They don't change during drt.

**Reply by Mohamed-Mejri** [reply]
Ah sorry I wasn't clear with what I said. I meant is it possible to get the metadata / state of the Graph/3D grid along side with the iRoutes and the global router guide after being preprocessed with the merging, splitting and bridging techniques. 

I want to gain from your processing of the inputs and just then try to export them if possible to any readable/parsable form (e.g JSON).

**Comment by maliberty** [comment, marked as answer]
The post-processed guides are generated as output_guide.mod in ORFS (controlled by -output_guide).   Other internal state of the router is not exported today.

**Reply by Mohamed-Mejri** [reply]
Thank you for clearing that up. One other question, is there any documentation / explanation for the structure of the output_guide.mod file?

**Comment by maliberty** [comment]
Its the same guide format as the input guide file

**Comment by Mohamed-Mejri** [comment]
Hello again,
I would like to get the access points to each pin in a net and I saw [this on another discussion](https://github.com/The-OpenROAD-Project/OpenROAD/discussions/625?sort=new#discussioncomment-636424), however, when I tried it the coordinates didn't make sense.
For example when I try it on the nangate45 gcd, for the _000_ net we have:  `_000_ ( _672_ D ) ( _504_ ZN ) + USE SIGNAL`
and the components in the net are  `_672_ DFF_X2 + PLACED ( 107540 145600 ) N ;`  and `_504_ OAI21_X1 + PLACED ( 107540 148400 ) FS ;`
and this is the output of the code
net _000_ 
( 46930 141260 ) layerNum 2 metal1 
( 46930 141120 ) layerNum 2 metal1
( 46740 141260 ) layerNum 2 metal1
( 51490 146580 ) layerNum 2 metal1
( 51490 146860 ) layerNum 2 metal1
( 51490 147140 ) layerNum 2 metal1
( 51490 147420 ) layerNum 2 metal1
( 50730 147420 ) layerNum 2 metal1
( 51870 147420 ) layerNum 2 metal1
( 51870 147700 ) layerNum 2 metal1
( 51870 147980 ) layerNum 2 metal1
( 51110 147420 ) layerNum 2 metal1

So from what I see these coordinates can't be correct, any idea why is this happening?

**Reply by maliberty** [reply]
Access points are now stored in odb and you can access them through the database APIs.  It is no longer necessary to hack the code to get at them.

**Reply by Mohamed-Mejri** [reply]
Thank you for your help!
